pkg/webhooks/workload_webhook_test.go

```diff
@@ -369,6 +371,208 @@ func TestValidateWorkloadUpdate(t *testing.T) {
+					testingutil.MakeAdmission("cluster-queue").
+						PodSets(kueue.PodSetAssignment{
+							Name: "ps1",
+							TopologyAssignment: &kueue.TopologyAssignment{
+								Levels: []string{"level"},
+								Domains: []kueue.TopologyDomainAssignment{
+									{
+										Values: []string{"abc"},
+										Count:  3,
+									},
+								},
+							},
+						}).
+						Obj(),
+				).
+				Obj(),
+			wantErr: field.ErrorList{
+				field.Invalid(field.NewPath("status", "admission"), nil, ""),
+			},
+		},
+		"TopologyAssignment can be set if delayedTopologyRequest is Pending": {
+			enableTopologyAwareScheduling: true,
+			before: testingutil.MakeWorkload(testWorkloadName, testWorkloadNamespace).
+				PodSets(
+					*testingutil.MakePodSet("ps1", 3).Obj(),
+				).
+				ReserveQuota(
+					testingutil.MakeAdmission("cluster-queue").
+						PodSets(kueue.PodSetAssignment{
+							Name:                   "ps1",
+							DelayedTopologyRequest: ptr.To(kueue.DelayedTopologyRequestStatePending),
+						}).
+						Obj(),
+				).
+				Obj(),
+			after: testingutil.MakeWorkload(testWorkloadName, testWorkloadNamespace).
+				PodSets(
+					*testingutil.MakePodSet("ps1", 3).Obj(),
+				).
+				ReserveQuota(
+					testingutil.MakeAdmission("cluster-queue").
+						PodSets(kueue.PodSetAssignment{
+							Name: "ps1",
+							TopologyAssignment: &kueue.TopologyAssignment{
+								Levels: []string{"level"},
+								Domains: []kueue.TopologyDomainAssignment{
+									{
+										Values: []string{"abc"},
+										Count:  3,
+									},
+								},
+							},
+						}).
+						Obj(),
+				).
+				Obj(),
+			wantErr: field.ErrorList{
+				field.Invalid(field.NewPath("status", "admission"), nil, ""),
+			},
+		},
+		"TopologyAssignment cannot be set if delayedTopologyRequest is Ready": {
+			enableTopologyAwareScheduling: true,
+			before: testingutil.MakeWorkload(testWorkloadName, testWorkloadNamespace).
+				PodSets(
+					*testingutil.MakePodSet("ps1", 3).Obj(),
+				).
+				ReserveQuota(
+					testingutil.MakeAdmission("cluster-queue").
+						PodSets(kueue.PodSetAssignment{
+							Name:                   "ps1",
+							DelayedTopologyRequest: ptr.To(kueue.DelayedTopologyRequestStateReady),
+						}).
+						Obj(),
+				).
+				Obj(),
+			after: testingutil.MakeWorkload(testWorkloadName, testWorkloadNamespace).
+				PodSets(
+					*testingutil.MakePodSet("ps1", 3).Obj(),
+				).
+				ReserveQuota(
+					testingutil.MakeAdmission("cluster-queue").
+						PodSets(kueue.PodSetAssignment{
+							Name: "ps1",
+							TopologyAssignment: &kueue.TopologyAssignment{
+								Levels: []string{"level"},
+								Domains: []kueue.TopologyDomainAssignment{
+									{
+										Values: []string{"abc"},
+										Count:  3,
+									},
+								},
+							},
+						}).
+						Obj(),
+				).
+				Obj(),
+			wantErr: field.ErrorList{
+				field.Invalid(field.NewPath("status", "admission"), nil, ""),
+			},
+		},
+		"PodSets cannot be added to admission": {
```

Comment:
```suggestion
		"PodSets cannot be removed from admission": {
```
Is this what you meant?